Improve ProtonVPN documentation

gene1wood opened the pull request:
* Remove the section that recommends users use follow the "custom provider" documentation. This section was accidentally left in, in https://github.com/qdm12/gluetun/pull/2390 but should have been removed (as @heronimoo [points out][1]). Removing this section will prevent users from following the "custom provider" instructions and incorrectly setting values like `WIREGUARD_ENDPOINT_IP`, `WIREGUARD_ENDPOINT_PORT`, `WIREGUARD_PUBLIC_KEY` and `WIREGUARD_ADDRESSES`. If the user sets `WIREGUARD_ENDPOINT_PORT` it triggers the ["endpoint port is set" error][2]
* Remove the note about each WireGuard server using it's own private key (as later in the page it states that the private key is the same for all servers)
* Expand the documentation about the `WIREGUARD_PRIVATE_KEY` value to state explicitly to ignore all the unused values in the ProtonVPN generated configuration file. This also changes the link from the ProtonVPN downloads page to the ProtonVPN page for generating WireGuard configs.
* Reorganize the VPN server port forwarding section to clarify which settings apply to OpenVPN and which apply to WireGuard.
* Add the `PORT_FORWARD_ONLY` setting to the VPN server port forwarding section
* Add examples to clarify the OpenVPN username appending trick for port forwarding and moderate NAT.
* Fix capitalization of WireGuard

[1]: https://github.com/qdm12/gluetun/pull/2390#issuecomment-2333917291
[2]: https://github.com/qdm12/gluetun/blob/68ddbfc0fed316f2e22c3b979b2186522a194da1/internal/configuration/settings/wireguardselection.go#L59-L64

TGM reviewed (commented):
LGTM

qdm12 commented on setup/providers/protonvpn.md:
````diff
@@ -11,7 +11,7 @@ docker run -it --rm --cap-add=NET_ADMIN --device /dev/net/run \
 
 ```sh
-# Wireguard
+# WireGuard
````
Unneeded change, reverting it:

```suggestion
# Wireguard
```

qdm12 commented on setup/providers/protonvpn.md:
```diff
@@ -46,10 +44,12 @@ services:
 - `OPENVPN_PASSWORD`
 
-### Wireguard only
+### WireGuard only
```
```suggestion
### Wireguard only
```

qdm12 commented on setup/providers/protonvpn.md:
```diff
@@ -46,10 +44,12 @@ services:
 
 - `VPN_TYPE=wireguard`
-- `WIREGUARD_PRIVATE_KEY` is your 32 bytes key in base64 format. The private key can only be obtained by [generating a Wireguard configuration file](https://account.protonvpn.com/downloads). Generate a Wireguard configuration file, copy the displayed `PrivateKey` value and optionally download the configuration file. Note this value is the same for all ProtonVPN servers. 💁 [Guide on how to generat ...
+- `WIREGUARD_PRIVATE_KEY` is your 32 byte key in base64 format.
+
+💁 To obtain the `WIREGUARD_PRIVATE_KEY` value, provision a configuration file from [account.proton.me/u/0/vpn/WireGuard](https://account.proton.me/u/0/vpn/WireGuard). The content of the WireGuard configuration file that ProtonVPN generates contains the `PrivateKey` value in the `[Interface]` section. Note that the private key value is the same for all ProtonVPN servers. All other values in the Pr ...
```
Only changed the wireguard config file generation link, since the rest of the paragraph is essentially just rewording the same thing imo

```suggestion
- `WIREGUARD_PRIVATE_KEY` is your 32 byte key in base64 format. The private key can be obtained by [generating a Wireguard configuration file](https://account.proton.me/u/0/vpn/WireGuard) and copy the displayed `PrivateKey` value. Note this value is the same for all ProtonVPN servers. 💁 [Guide on how to generate a configuration file](https://protonvpn.com/support/wireguard-configurations/)
```

qdm12 commented on setup/providers/protonvpn.md:
```diff
@@ -72,9 +72,12 @@ services:
-- Add `+pmp` to your OpenVPN username (thanks to [@mortimr](https://github.com/qdm12/gluetun/issues/1760#issuecomment-1669518288))
 - `VPN_PORT_FORWARDING=on`
-- If you use **Wireguard** using the custom provider, set `VPN_PORT_FORWARDING_PROVIDER=protonvpn`
+- `PORT_FORWARD_ONLY=on` to tell Gluetun to only connect to servers that support port forwarding
```
This is automated and no longer a requirement 😉 

```suggestion
```

qdm12 commented on setup/providers/protonvpn.md:
```diff
@@ -72,9 +72,12 @@ services:
+- `PORT_FORWARD_ONLY=on` to tell Gluetun to only connect to servers that support port forwarding
+- For OpenVPN
+  - Append `+pmp` to your OpenVPN username. For example, if your ProtonVPN username was `johndoe`, set `OPENVPN_USER` to `johndoe+pmp` (thanks to [@mortimr](https://github.com/qdm12/gluetun/issues/1760#issuecomment-1669518288)). ProtonVPN documentation on [port forwarding with OpenVPN](https://protonvpn.com/support/port-forwarding-manual-setup#openvpn)
+- For WireGuard
+  - Set `VPN_PORT_FORWARDING_PROVIDER=protonvpn`
```
That's no longer needed 😉 This is only needed if you use the custom provider.

```suggestion
```

qdm12 commented on setup/providers/protonvpn.md:
```diff
@@ -72,9 +72,12 @@ services:
 - `VPN_PORT_FORWARDING=on`
-- If you use **Wireguard** using the custom provider, set `VPN_PORT_FORWARDING_PROVIDER=protonvpn`
+- `PORT_FORWARD_ONLY=on` to tell Gluetun to only connect to servers that support port forwarding
+- For OpenVPN
+  - Append `+pmp` to your OpenVPN username. For example, if your ProtonVPN username was `johndoe`, set `OPENVPN_USER` to `johndoe+pmp` (thanks to [@mortimr](https://github.com/qdm12/gluetun/issues/1760#issuecomment-1669518288)). ProtonVPN documentation on [port forwarding with OpenVPN](https://protonvpn.com/support/port-forwarding-manual-setup#openvpn)
```
I prefer it shorter, but nice example still 👍 Also nice `Add` replacement with `Append`!

```suggestion
- For OpenVPN only, append `+pmp` to your OpenVPN username (thanks to [@mortimr](https://github.com/qdm12/gluetun/issues/1760#issuecomment-1669518288)). If needed, see the [ProtonVPN OpenVPN port forwarding documentation](https://protonvpn.com/support/port-forwarding-manual-setup#openvpn).
```

qdm12 commented on setup/providers/protonvpn.md:
```diff
@@ -86,7 +89,9 @@ For example setting `SERVER_HOSTNAMES=ch-us-01a.protonvpn.com` would set a multi
 Paid ProtonVPN subscribers can optionally use [Moderate NAT](https://protonvpn.com/support/moderate-nat/) on their connections.
 
-To do so, the OpenVPN username assigned by ProtonVPN should have `+nr` appended to the end of it.
+To do so with OpenVPN, append `+nr` to the ProtonVPN username. For example, if your ProtonVPN username was `johndoe`, set `OPENVPN_USER` to `johndoe+nr`.
+
+To do so with WireGuard, when provisioning a configuration file from [account.proton.me/u/0/vpn/WireGuard](https://account.proton.me/u/0/vpn/WireGuard), under `Select VPN options`, enable `Moderate NAT` before creating your configuration file containing your private key.
```
```suggestion
- OpenVPN: append `+nr` to your ProtonVPN username
- WireGuard: when [generating a configuration file](https://account.proton.me/u/0/vpn/WireGuard), under `Select VPN options`, enable `Moderate NAT`
```

qdm12 reviewed (commented):
Thanks! 💯

qdm12 commented on setup/providers/protonvpn.md:
```diff
@@ -49,7 +47,7 @@ services:
 
 - `VPN_TYPE=wireguard`
-- `WIREGUARD_PRIVATE_KEY` is your 32 bytes key in base64 format. The private key can only be obtained by [generating a Wireguard configuration file](https://account.protonvpn.com/downloads). Generate a Wireguard configuration file, copy the displayed `PrivateKey` value and optionally download the configuration file. Note this value is the same for all ProtonVPN servers. 💁 [Guide on how to generat ...
+- `WIREGUARD_PRIVATE_KEY` is your 32 byte key in base64 format. The private key can be obtained by [generating a Wireguard configuration file](https://account.proton.me/u/0/vpn/WireGuard) and copy the displayed `PrivateKey` value. Note this value is the same for all ProtonVPN servers. 💁 [Guide on how to generate a configuration file](https://protonvpn.com/support/wireguard-configurations/)
```
Interesting `32 bytes key` -> `32 byte key` - TIL about byte being an adjective and not a quantity!

qdm12 reviewed (commented):
Summary:
- Migrate Wireguard instructions from using the custom provider to be native
- Update Wireguard configuration generation url
- Add how to use moderate nat for Wireguard

gene1wood commented:
@qdm12 

The reason I had added the sentence 

> All other values in the ProtonVPN WireGuard Configuration file should be ignored and not set in Gluetun environment variables.

is because a number of the variables [in the Gluetun Wireguard setup documentation](https://github.com/qdm12/gluetun-wiki/blob/main/setup/providers/custom.md#wireguard-only) are present in the ProtonVPN generated configuration, however if you add those Gluetun environment variables (which to me seemed intuitive because the Gluetun Wireguard setup page describes them), then Gluetun doesn't work with ProtonVPN. 

My goal was to clarify the confusion the blocked me from getting things working which is that you *must* only add the `PrivateKey` value from the ProtonVPN and nothing else.

I added the examples of what it means to append `+pmp` and `+nr` to the username 

> For example, if your ProtonVPN username was `johndoe`, set `OPENVPN_USER` to `johndoe+pmp`

because when I first set this up, I didn't understand it from the Gluetun documentation and had to do a lot of digging in the Proton VPN documentation to understand what this meant. I thought that an example would make it clear.

I should have clarified in the description of this PR *why* I was adding the note to avoid carrying over any other wireguard settings from the ProtonVPN config to Gluetun and *why* I was adding the examples for the username appending trick.

Given these reasons, would you like me to open a PR with just those additions? I think it would help users (like me) who get stuck trying to setup ProtonVPN.
